lib/response-handler.js

```diff
@@ -86,7 +86,7 @@ module.exports = (bluebird, error, logger, parseLinkHeader, URI) => {
 
         uri = new URI(link, url);
 
-        return uri.toString();
+        return uri.path().toString();
```

Comment:
I see that you have added this elsewhere, but I think it MUST either 1. go in this function or 2. the places where resolveLink is used directly needs to be updated to use something else.  Remember, the below script must work.

```
#!/usr/bin/env node
"use strict";

var artifact, newArtifact, shelfLib, ref;

shelfLib = require("shelf-lib")("https://api.shelf-qa.cwscloud.net", {strictHostCheck: false});
ref = shelfLib.initReference("morty", "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX");
artifact = ref.initArtifactWithTimestamp("/shelf-js-test/an-artifact");
artifact.uploadFromFile("test-artifact.txt").then((loc) => {
    console.log("LOCATION:", loc);
    newArtifact = ref.initArtifact(loc);
    newArtifact.download().then((content) => {
        console.log("GOOD NEWS, we were able to download content:", content);
    });
}, (err) => {
    console.log("OH NO ERROR", err.message);
});
```